### PR TITLE
[KYUUBI #842] Fix meet rat check error when build kyuubi project

### LINKED ISSUE #842: Fix meet rat check error when build kyuubi project
### 1. Describe the bug

Meet rat check error when build kyuubi project
```
[ERROR] Failed to execute goal org.apache.rat:apache-rat-plugin:0.13:check (default-cli) on project kyuubi: Too many files with unapproved license: 75 See RAT report in: /work/projects/opensource/kyuubi/target/rat.txt -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
```

### 2. Steps to reproduce

1. Build kyuubi project
```
./build/dist --tgz --spark-provided -Pkyuubi-extension-spark_3.1
```
2. Untar dist package and check some jars in packages
3. Build kyuubi again
4. Met the rat check error

### PR DESCRIPTION
**Pull request by timothy65535**
Fixes #842

### 1. Describe the bug

Meet rat check error when build kyuubi project
```
[ERROR] Failed to execute goal org.apache.rat:apache-rat-plugin:0.13:check (default-cli) on project kyuubi: Too many files with unapproved license: 75 See RAT report in: /work/projects/opensource/kyuubi/target/rat.txt -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
```

### 2. Steps to reproduce

1. Build kyuubi project
```
./build/dist --tgz --spark-provided -Pkyuubi-extension-spark_3.1
```
2. Untar dist package and check some jars
3. Build kyuubi again

### _How was this patch tested?_
- [X] Add some test cases that check the changes thoroughly including negative and positive cases if possible

```
mvn org.apache.rat:apache-rat-plugin:0.13:check
```

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

image

```
74 Unknown Licenses

*****************************************************

Files with unapproved licenses:

  kyuubi-1.3.0-SNAPSHOT-bin-spark-3.1-hadoop3.2/externals/spark-3.1.2-bin-hadoop3.2/data/mllib/ridge-data/lpsa.data
  kyuubi-1.3.0-SNAPSHOT-bin-spark-3.1-hadoop3.2/externals/spark-3.1.2-bin-hadoop3.2/data/mllib/als/test.data
  kyuubi-1.3.0-SNAPSHOT-bin-spark-3.1-hadoop3.2/externals/spark-3.1.2-bin-
  kyuubi-1.3.0-SNAPSHOT-bin-spark-3.1-hadoop3.2/externals/spark-3.1.2-bin-hadoop3.2/R/lib/SparkR/doc/sparkr-vignettes.R
  kyuubi-1.3.0-SNAPSHOT-bin-spark-3.1-hadoop3.2/externals/spark-3.1.2-bin-hadoop3.2/R/lib/SparkR/doc/index.html
  kyuubi-1.3.0-SNAPSHOT-bin-spark-3.1-hadoop3.2/externals/spark-3.1.2-bin-hadoop3.2/R/lib/SparkR/R/SparkR
  kyuubi-1.3.0-SNAPSHOT-bin-spark-3.1-hadoop3.2/RELEASE
  kyuubi-1.3.0-SNAPSHOT-bin-without-spark/RELEASE

*****************************************************

```

### CODE REVIEW

**Comment by yaooqinn** [comment]
I don't think we need to exclude files added manually.

Maybe we shall add a LICENSE head to `kyuubi-1.3.0-SNAPSHOT-bin-without-spark/RELEASE`?

File: pom.xml [hunk 1]
```diff
@@ -1371,6 +1371,8 @@
                             <exclude>build/scala-*/**</exclude>
                             <exclude>**/**/operation_logs/**/**</exclude>
                             <exclude>**/*.output.schema</exclude>
+                            <exclude>kyuubi-*-bin-without-spark/**</exclude>
+                            <exclude>kyuubi-*-bin-spark-*/**</exclude>
```
Comment (pan3793) [inline pom.xml:1375]: we can merge them to `kyuubi-*-bin-*/**`

**Comment by pan3793** [comment]
> I don't think we need to exclude files added manually.

+1. The `./build/dist` without `--tgz` will output to `dist`, which excluded by rat. The unarchived folder is out of the scope.

> Maybe we shall add a LICENSE head to `kyuubi-1.3.0-SNAPSHOT-bin-without-spark/RELEASE`

Spark also does not add LICENSE head in RELEASE.

**Comment by yaooqinn** [comment]
> > I don't think we need to exclude files added manually.
> 
> +1. The `./build/dist` without `--tgz` will output to `dist`, which excluded by rat. The unarchived folder is out of the scope.
> 
> > Maybe we shall add a LICENSE head to `kyuubi-1.3.0-SNAPSHOT-bin-without-spark/RELEASE`
> 
> Spark also does not add LICENSE head in RELEASE.

Can we check other Apache Incubating Projects which have some releases came out. The Apache Policy is evolving.

**Comment by yaooqinn** [comment]
`-without-spark` can be ommited

**Comment by pan3793** [comment]
Have not seen another project that has a RELEASE file in binary dist. It should be acceptable that some doc files like `CHANGELOG.md` `README.md` dist without licence head. (ref: Skywalking)

**Review by yaooqinn — APPROVED** [review]

**Comment by timothy65535** [comment]
Add `**` as prefix, it suits more scenarios.
```
   <exclude>**/kyuubi-*-bin-without-spark/**</exclude>
   <exclude>**/kyuubi-*-bin-spark-*/**</exclude>
```

**Comment by ulysses-you** [comment]
thanks, merging to master for v1.3.0

**Pull request closed without merging**